src/marten/db/query/set.cr

```diff
@@ -325,6 +325,40 @@ module Marten
           add_query_node(query_node)
         end
 
+        # Returns if a query set includes a specific model.
+        def includes?(value : DB::Model)
+          query_node = Node.new({Constants::PRIMARY_KEY_ALIAS => value.pk})
+          add_query_node(query_node)
+          exists?
```

Comment:
Note that `#add_query_node` returns a _new_ query set object (query sets are _never_ mutated). So presently, the `#exists?` call is executed on the initial (and unfiltered) query set.

Another thing to take into account is that the query set could already be evaluated when this method is executed: in this situation, we can avoid hitting the DB again by simply calling `#includes?` on the cached array of records.

To fix all this, you could simply do something like this:

```suggestion
          if @result_cache.nil?
            filter(Node.new({Constants::PRIMARY_KEY_ALIAS => value.pk})).exists?
          else
             @result_cache.not_nil!.includes?(value)
           end
```